coderd/coderd.go

```diff
@@ -208,6 +211,12 @@ func New(options *Options) *API {
 	if options == nil {
 		options = &Options{}
 	}
+	if options.NewTicker == nil {
+		options.NewTicker = func(duration time.Duration) (tick <-chan time.Time, done func()) {
+			ticker := time.NewTicker(duration)
+			return ticker.C, ticker.Stop
+		}
+	}
```

Comment:
This is OK for now but we should probably swap this out for something better later on 🕰️

Reply:
I agree. Some fake clock solution would be ideal, then we can use across all tests more generally.